connector/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -0,0 +1,206 @@
+    private static Optional<String> createWhereClause(String[] filters) {
+        return Optional.empty();
+    }
+
+    public TableInfo getTable(TableId tableId) {
+        TableId bigQueryTableId = tableIds.get(tableId);
+        Table table = bigQuery.getTable(bigQueryTableId != null ? bigQueryTableId : tableId);
+        if (table != null) {
+            tableIds.putIfAbsent(tableId, table.getTableId());
+            datasetIds.putIfAbsent(toDatasetId(tableId), toDatasetId(table.getTableId()));
+        }
+        return table;
+    }
+
+    public TableInfo getSupportedTable(TableId tableId, boolean viewsEnabled, String viewEnabledParamName) {
+        TableInfo table = getTable(tableId);
+        if (table == null) {
+            return null;
+        }
+
+        TableDefinition tableDefinition = table.getDefinition();
+        TableDefinition.Type tableType = tableDefinition.getType();
+        if (TableDefinition.Type.TABLE == tableType) {
+            return table;
+        }
+        if (TableDefinition.Type.VIEW == tableType) {
+            if (viewsEnabled) {
+                return table;
+            } else {
+                throw new BigQueryConnectorException(UNSUPPORTED, format(
+                        "Views are not enabled. You can enable views by setting '%s' to true. Notice additional cost may occur.",
+                        viewEnabledParamName));
+            }
+        }
+        // not regular table or a view
+        throw new BigQueryConnectorException(UNSUPPORTED, format("Table type '%s' of table '%s.%s' is not supported",
+                tableType, table.getTableId().getDataset(), table.getTableId().getTable()));
+    }
+
+    DatasetId toDatasetId(TableId tableId) {
+        return DatasetId.of(tableId.getProject(), tableId.getDataset());
+    }
+
+    String getProjectId() {
+        return bigQuery.getOptions().getProjectId();
+    }
+
+    Iterable<Dataset> listDatasets(String projectId) {
+        final Iterator<Dataset> datasets = bigQuery.listDatasets(projectId).iterateAll().iterator();
+        return () -> Iterators.transform(datasets, this::addDataSetMappingIfNeeded);
+    }
+
+    Iterable<Table> listTables(DatasetId datasetId, TableDefinition.Type... types) {
+        Set<TableDefinition.Type> allowedTypes = ImmutableSet.copyOf(types);
+        DatasetId bigQueryDatasetId = datasetIds.getOrDefault(datasetId, datasetId);
+        Iterable<Table> allTables = bigQuery.listTables(bigQueryDatasetId).iterateAll();
+        return StreamSupport.stream(allTables.spliterator(), false)
+                .filter(table -> allowedTypes.contains(table.getDefinition().getType()))
+                .collect(toImmutableList());
+    }
+
+    private Dataset addDataSetMappingIfNeeded(Dataset dataset) {
+        DatasetId bigQueryDatasetId = dataset.getDatasetId();
+        DatasetId prestoDatasetId = DatasetId.of(bigQueryDatasetId.getProject(), bigQueryDatasetId.getDataset().toLowerCase(ENGLISH));
```

Comment:
I don't understand the relation between presto and bigquery datasetIds here

Reply:
Should have been removed